### ISSUE TITLE
Error with top-level await

### ISSUE THREAD

**Issue by jsulpis** [CodyJasonBennett/four#5]
Hi,
thank you for the lib, it looks promising !

I tried to use it for a basic WebGL test but ran into the following issue : 

![image](https://github.com/CodyJasonBennett/four/assets/22420399/59ca5974-9eb1-4a3d-a286-6b21f85e8f15)

Indeed, I see that there are two top-level constants `_adapter` and `_device` defined in `WebGPURenderer.ts` that await a promise. The error pops up even when using WebGL, I guess because the exported file `index.js` bundles everythings.

I tried a quick fix, but since the request of the device is async, I don't understand how the creation of the WebGPURenderer can not be async. I could work on a PR if you have an idea of how you want to address that

**Comment by CodyJasonBennett** [comment]
This comes from Vite's `optimizeDeps`. A quick fix might be to split the renderers into separate files (bundlers skip unused paths).

In your case, you can tweak the target with:

```js
// vite.config.js
optimizeDeps: {
  esbuildOptions: {
    target: 'esnext'
  }
}
```

It's a bit unfortunate Vite/ESBuild/browserslist is pedantic about this since ESM spec guarantees the availability of top-level await which is why this is published as such.

**Comment by jsulpis** [comment]
Thank you for the explanation and the workaround !

Your config fixes the issue in dev mode but not during the build. Based on [this issue](https://github.com/vitejs/vite/issues/6985) I found that setting `build.target` to `esnext` fixes the issue both in dev and build mode. 

But I ended up using the proposed plugin [vite-plugin-top-level-await](https://github.com/Menci/vite-plugin-top-level-await) in order to have a wider browser support.

In any case, maybe a workaround should be proposed in the README ? I think many users will be using Vite.

As everything works fine for me now, feel free to close the issue if you think it's ok on your side.
Thank you again !
